lib/src/app/app.dart

```diff
@@ -16,25 +17,151 @@ class AppClient {
+
+  /// Create a new [Organization]
+  Future<Organization> createOrganization(String name) async {
+    final request = CreateOrganizationRequest()..name = name;
+    final CreateOrganizationResponse response = await _client.createOrganization(request);
+    return response.organization;
+  }
+
   /// List all the [Organization] the currently authenticated user has access to
   Future<List<Organization>> listOrganizations() async {
     final listOrganizationsRequest = ListOrganizationsRequest();
     final ListOrganizationsResponse response = await _client.listOrganizations(listOrganizationsRequest);
     return response.organizations;
   }
 
+  /// Get all [OrganizationIdentity]s that have access to a [Location].
+  Future<List<OrganizationIdentity>> getOrganizationsWithAccessToLocation(String locationId) async {
+    final request = GetOrganizationsWithAccessToLocationRequest();
+    final GetOrganizationsWithAccessToLocationResponse response = await _client.getOrganizationsWithAccessToLocation(request);
+    return response.organizationIdentities;
+  }
+
+  /// List the [Organization]s a user belongs to
+  Future<List<OrgDetails>> listOrganizationsByUser(String userId) async {
+    final request = ListOrganizationsByUserRequest()..userId = userId;
+    final ListOrganizationsByUserResponse response = await _client.listOrganizationsByUser(request);
+    return response.orgs;
+  }
+
   /// Get a specific [Organization] by ID
   Future<Organization> getOrganization(String organizationId) async {
     final getOrganizationRequest = GetOrganizationRequest()..organizationId = organizationId;
     final GetOrganizationResponse response = await _client.getOrganization(getOrganizationRequest);
     return response.organization;
   }
 
-  /// List the [Location] of a specific [Organization] that the currently authenticated user has access to
-  Future<List<Location>> listLocations(Organization organization) async {
-    final listLocationsRequest = ListLocationsRequest()..organizationId = organization.id;
-    final ListLocationsResponse response = await _client.listLocations(listLocationsRequest);
-    return response.locations;
+  /// Checks for namespace availablity throughout all [Organization]s.
+  Future<bool> getOrganizationNamespaceAvailability(String publicNamespace) async {
+    final request = GetOrganizationNamespaceAvailabilityRequest()..publicNamespace = publicNamespace;
+    final GetOrganizationNamespaceAvailabilityResponse response = await _client.getOrganizationNamespaceAvailability(request);
+    return response.available;
+  }
+
+  /// Update an [Organization]
+  Future<Organization> updateOrganization(String organizationId,
+      {String? name, String? publicNamespace, String? region, String? cid}) async {
+    final request = UpdateOrganizationRequest()
+      ..organizationId = organizationId
+      ..name = name ?? ''
+      ..publicNamespace = publicNamespace ?? ''
+      ..region = region ?? ''
+      ..cid = cid ?? '';
```

Comment:
I _think_ instead we should be doing, e.g., 
```dart
final request = UpdateOrganizationRequest();
if (name != null) request.name = name;
```

My fear is that by setting to an empty string, on the app side these values will get overridden by an empty string (as opposed to left untouched).